File: App/src/app/_core/_model/account.ts
import { AccountGroupAccount } from "./account-group-account";
import { AccountType } from "./account.type";

export interface Account {
  id: number;
  username: string;
  fullName: string;
  password: string;
  email: string;
  isLock: boolean;
  accountTypeId: number | null;
  factId: number | null;
  centerId: number | null;
  deptId: number | null;
  createdBy: number;
  modifiedBy: number | null;
  createdTime: string;
  modifiedTime: string | null;
  accountType: AccountType;
  accountGroupText: string;
  accountGroupIds: number[];
  leader: number;
  manager: number;
  leaderName: string;
  managerName: string;
  l1: number | null;
  l1Name: string;
  l2: number | null;
  l2Name: string;
  functionalLeader: number | null;
  functionalLeaderName: string;
  l0: boolean;
  ghr: boolean;
  jobTitleId: number | null;
  gm: boolean;
  gmScore: boolean;
  systemFlow: number | null;
}

File: A4KPI/Models/Account.cs
﻿using A4KPI.Models.Interface;
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace A4KPI.Models
{
    [Table("Accounts")]
    public class Account: IDateTracking
    {
        [Key]
        public int Id { get; set; }
        [MaxLength(255)]
        public string Username { get; set; }
        [MaxLength(255)]
        public string FullName { get; set; }
        [MaxLength(255)]
        public string Password { get; set; }
        [MaxLength(255)]
        public string Email { get; set; }
        public bool IsLock { get; set; }
        public int? AccountTypeId { get; set; }
        public int? FactId { get; set; }
        public int? CenterId { get; set; }
        public int? DeptId { get; set; }
        public int? Leader { get; set; }
        public int? Manager { get; set; }
        public int CreatedBy { get; set; }
        public int? ModifiedBy { get; set; }
        public DateTime CreatedTime { get ; set ; }
        public DateTime? ModifiedTime { get ; set ; }
        public bool? L0 { get; set; }
        public int? L1 { get; set; }
        public int? L2 { get; set; }
        public int? FunctionalLeader  { get; set; }
        public bool? GHR { get; set; }
        public int? JobTitleId { get; set; }
        public bool? GM { get; set; }
        public bool? GMScore { get; set; }
        public int? SystemFlow { get; set; }

        [ForeignKey(nameof(AccountTypeId))]
        public virtual AccountType AccountType { get; set; }

        public virtual ICollection<AccountGroupAccount> AccountGroupAccount { get; set; }


    }
}
